proto/backend.proto

```diff
@@ -15,16 +15,27 @@ message DataSourceConfig {
   string database = 5;
   bool basicAuthEnabled = 6;
   string basicAuthUser = 7;
+
+  // from [data_source.json_data] field in the database
+  bytes jsonData = 8;
+  
+  // from [data_source.secure_json_data] field in the database
+  map<string,string> decryptedSecureJsonData = 9;
+  int64 lastUpdatedMS = 10;
 }
 
 message PluginConfig {
   int64 orgId = 1;
   string pluginId = 2;
-  string pluginType = 3;
-  bytes jsonData = 4;
-  map<string,string> decryptedSecureJsonData = 5;
-  int64 updatedMS = 6;
-  DataSourceConfig datasourceConfig = 7;
+  
+  // from [plugin_setting.json_data] field in the database
+  bytes jsonData = 3;
+  
+  // from [plugin_setting.secure_json_data] field in the database
+  map<string,string> decryptedSecureJsonData = 4;
+  int64 lastUpdatedMS = 5;
+
+  DataSourceConfig datasourceConfig = 6;
```

Comment:
Having `datasourceConfig` in `PluginConfig` and both have `jsonData`, `decryptedSecureJsonData` and `lastUpdatedMS` makes it more unclear for me and from a plugin developer perspective weird that they need to adapt to how data are stored in Grafana database.

Reply:
Do you think it would be more clear if DatasourceConfig and PluginConfig was at the same level?

Ex. 
```protobuf
message DataQueryRequest {
  // Plugin Configuration
  PluginConfig pluginConfig = 1;

  // Datasource Configuration
  DatasourceConfig datasourceConfig = 2;

  // Environment info
  map<string,string> headers = 2;

  // List of queries
  repeated DataQuery queries = 3;

  //Info about the user who calls the plugin.
  User user = 4;
}
```

Reply:
> from a plugin developer perspective weird that they need to adapt to how data are stored in Grafana database.

Since jsondata can be stored in plugin_settings(same value for whole plugin) and at datasource (different for each datasource) I dont see how we cannot have two fields. Naming could be improved but I still think we need two fields.

Reply:
Maybe, but then I would expect pluginConfig to be populated and not datasourceConfig or vice versa since jsonData, decryptedSecureJsonData and lastUpdatedMS could/should only be populated in one of them. Having one PluginConfig makes all a lot easier. 

You can see some history how it looked before `DatasourceConfig` was added and we only had one `PluginConfig`. Maybe you prefer to revert back to that? https://github.com/grafana/grafana-plugin-sdk-go/pull/54/files#diff-ad0dcffdf701a0711096c6d383ad6cd5

:confused:

Reply:
> Since jsondata can be stored in plugin_settings(same value for whole plugin) and at datasource (different for each datasource) I dont see how we cannot have two fields. Naming could be improved but I still think we need two fields.

Wait what? Do you mean that a data source can store stuff in plugin_setting??? Isn't that solely for app plugins? Why have we implemented it this way? I guess from Grafana UI you can only store stuff in plugin_setting for app plugins, right?